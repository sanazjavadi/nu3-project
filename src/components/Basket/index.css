.basket-container {
  position: relative;
  width: 50px;
  height: 50px;
  cursor: pointer;
}

.basket-count-badge {
  position: absolute;
  width: 20px;
  height: 20px;
  text-align: center;
  top: -1px;
  right: 10px;
  font-size: 0.7rem;
  color: antiquewhite;
  font-weight: bold;
  background: red;
  border-radius: 50%;
}
