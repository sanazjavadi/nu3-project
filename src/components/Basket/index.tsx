import React from "react";
import useCard from "../../hooks/useCard";
import "./index.css";
import { HiOutlineShoppingBag } from "react-icons/hi";

const Basket = () => {
  const { totalItemsCount } = useCard();
  return (
    <div
      className="basket-container flex items-center"
      data-testid="basket-container"
    >
      <span className="basket-count-badge">{totalItemsCount}</span>
      <HiOutlineShoppingBag />
    </div>
  );
};

export default Basket;
