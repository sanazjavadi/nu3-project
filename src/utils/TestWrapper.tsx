import { configureStore, Store } from "@reduxjs/toolkit";
import { MemoryRouter } from "react-router-dom";
import React, { ComponentType } from "react";
import { Provider } from "react-redux";
import reducer from "../reducers";

export const makeStore = (): Store => {
  return configureStore({ reducer });
};

const wrapComponent = (
  Component: ComponentType,
  store: Store | null = null,
  props = {}
): JSX.Element => {
  return (
    <Provider store={store || makeStore()}>
      <MemoryRouter>
        <Component {...props} />
      </MemoryRouter>
    </Provider>
  );
};

export default wrapComponent;
