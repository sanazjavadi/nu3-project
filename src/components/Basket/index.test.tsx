import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TestWrapper, { makeStore } from "../../utils/TestWrapper";
import Basket from "./";

const store = makeStore();

describe("Basket Tests Cases", () => {
  test("AppBasket: Renders", () => {
    render(TestWrapper(Basket, store, { productsCount: 1 }));
    const basketElement = screen.getByTestId("basket-container");
    expect(basketElement).toBeInTheDocument();
  });

  test("AppBasket: Show Product Count In Basket", () => {
    render(TestWrapper(Basket, store, { productsCount: 1 }));
    const basketElement = screen.getByTestId("basket-container");
    expect(basketElement).toContainHTML("1");
  });
});

test("AppBasket: Snapshot", () => {
  const component = renderer.create(
    TestWrapper(Basket, store, { productsCount: 1 })
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
