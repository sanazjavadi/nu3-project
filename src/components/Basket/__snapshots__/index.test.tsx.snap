// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader: Snapshot 1`] = `
<div
  className="basket-container flex items-center"
  data-testid="basket-container"
>
  <span
    className="basket-count-badge"
    data-testid="basket-count"
  >
    1
  </span>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 11V7a4 4 0 00-8 0v4M5 9h14l1 12H4L5 9z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
</div>
`;
